src/main/java/org/elasticsearch/plugins/PluginManager.java

```diff
@@ -19,674 +19,557 @@
-            sc.init(null, trustAllCerts, new java.security.SecureRandom());
-            HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
-        } catch (Exception e) {
-            throw new ElasticsearchException("Failed to install all-trusting trust manager", e);
+            }
         }
     }
 
-    public void downloadAndExtract(String name) throws IOException {
-        if (name == null) {
-            throw new ElasticsearchIllegalArgumentException("plugin name must be supplied with --install [name].");
-        }
-        HttpDownloadHelper downloadHelper = new HttpDownloadHelper();
-        boolean downloaded = false;
-        HttpDownloadHelper.DownloadProgress progress;
-        if (outputMode == OutputMode.SILENT) {
-            progress = new HttpDownloadHelper.NullProgress();
-        } else {
-            progress = new HttpDownloadHelper.VerboseProgress(SysOut.getOut());
+    /**
+     * List all installed plugins
+     */
+    static class List extends CliTool.Command {
+
+        private static final String NAME = "list";
+
+        private static final CliToolConfig.Cmd CMD = cmd(NAME, List.class).build();
+
+        public static Command parse(Terminal terminal, CommandLine cli) {
+            return new List(terminal);
         }
 
-        if (!Files.isWritable(environment.pluginsFile())) {
-            throw new IOException("plugin directory " + environment.pluginsFile() + " is read only");
+        List(Terminal terminal) {
+            super(terminal);
         }
 
-        PluginHandle pluginHandle = PluginHandle.parse(name);
-        checkForForbiddenName(pluginHandle.name);
+        @Override
+        public ExitStatus execute(Settings settings, Environment env) throws Exception {
+            Collection<PluginDescriptor> plugins = installedPluginsRepository(env, settings).list();
+            if (plugins != null) {
+                for (PluginDescriptor plugin : plugins) {
+                    terminal.println("%s (%s)", plugin.name(), checkVersion(plugin.version()));
+                    terminal.println(VERBOSE, "\torganisation: %s", plugin.organisation());
+                    terminal.println(VERBOSE, "\tname        : %s", plugin.name());
+                    terminal.println(VERBOSE, "\tversion     : %s", plugin.version());
+                    terminal.println(VERBOSE, "\tdescription : %s", plugin.description());
 
-        Path pluginFile = pluginHandle.distroFile(environment);
-        // extract the plugin
-        final Path extractLocation = pluginHandle.extractedDir(environment);
-        if (Files.exists(extractLocation)) {
-            throw new IOException("plugin directory " + extractLocation.toAbsolutePath() + " already exists. To update the plugin, uninstall it first using --remove " + name + " command");
-        }
+                    Set<PluginDescriptor> dependencies = plugin.dependencies();
+                    if ((dependencies != null) && (!dependencies.isEmpty())) {
```

Comment:
I think we don't need to check for `null` here. We are using `dependencies.build()` to create this set and it can't generate `null`.